src/Harmony/Signer.cpp

```diff
@@ -441,8 +440,9 @@ Data Signer::rlpNoHashDirective(const Staking<EditValidator> &transaction) const
     append(encoded, RLP::encodeList(descriptionEncoded));
 
     auto decEncoded = Data();
-    if (transaction.stakeMsg.commissionRate != nullptr) {
-        append(decEncoded, RLP::encode(transaction.stakeMsg.commissionRate->value));
+    if (transaction.stakeMsg.commissionRate.has_value()) {
+        // Note: std::optional.value() is not available in XCode with target < iOS 12; using '*'
```

Comment:
probably time to only support iOS 12 +

Reply:
For this particular issue it's not a big deal, instead of optional.value() alternative is *optional.

Reply:
our app supports iOS 13+, now iOS 14 comes out, iOS 12/13/14 should be good enough, besides we had at least 3 times build failure because of `optional.value()`